### ISSUE TITLE
Classes applied in wrong order.

### ISSUE THREAD

**Issue by Gisleburt** [styled-components/styled-components#2716]
We're experiencing the same problem as listed in #1164 of classes being in the wrong order in one project consistently.

We have a styleguide that we import into multiple projects, all but one of which are fine, and the one that is wrong is consistently wrong, there's no randomness to it, and it only appears to be broken on one component, but it might be the only one where the defaults are being overridden (we extend/restyle other components but usually for margin to position them inside something else). I can't see anything special about the component.

The styleguide all consuming projects use styled-components `4.3.2`. The only quirk I can think of is that the one that is broken is attached to a shadow-dom.

We are using the `styled(MyComponent)` feature. Reading #1164 I was hoping the `.extend` might fix the problem by simply creating a new class but it seems that feature was removed.

## Environment

```
## System:
 - OS: macOS 10.14.6
 - CPU: (4) x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
 - Memory: 1.41 GB / 8.00 GB
 - Shell: 5.3 - /bin/zsh
## Binaries:
 - Node: 12.6.0 - /usr/local/bin/node
 - Yarn: 1.17.3 - /usr/local/bin/yarn
 - npm: 6.10.3 - /usr/local/bin/npm
 - Watchman: 4.9.0 - /usr/local/bin/watchman
## npmPackages:
 - styled-components: ^4.3.2 => 4.3.2 
```

## Reproduction

Our styleguide: https://gitlab.com/apolitical/styleguide
The effected project https://gitlab.com/apolitical/frontend-partials

The effected component is the ProfilePicture: https://gitlab.com/apolitical/styleguide/blob/master/src/Component/Atom/ProfilePicture/ProfilePicture.jsx
And it is extended here: https://gitlab.com/apolitical/styleguide/blob/master/src/Component/Organism/DropDownProfileMenu.jsx#L61

The problem stems around which hight it uses, 3rem/4rem, or 100%

## Steps to reproduce

Unknown, it happens consitently in one project but no others.

## Expected Behavior

Classes should be applied in order of precedence

## Actual Behavior

They are not.

**Comment by probablyup** [comment]
This should be fixed in v4.4 / v5 latest beta.

**Comment by blakeplumb** [comment]
I am having the same problem in version 5.3.10.
